onnxruntime/core/session/provider_bridge_ort.cc

```diff
@@ -213,6 +253,122 @@ common::Status LoadDynamicLibraryFromProvider(onnxruntime::PathString library_na
 struct ProviderHostImpl : ProviderHost {
   const OrtApiBase* OrtGetApiBase() override { return ::OrtGetApiBase(); }
 
+  Status GetEPOptimizerByName(const std::string& name,
+                              const GraphTransformerManager& transformer_mgr,
+                              std::function<std::vector<std::unique_ptr<ComputeCapability>>(const GraphViewer&)>& selection_func) override {
+    static const GraphTransformerManager& graph_transformer_mgr = transformer_mgr;
+    std::string optimizer_name(name);
+
+    // Pre-defined graph transformers/optimizers
+    static const std::string kEP_GRAPH_TRANSFORMER_CONSTANT_FOLDING_DQ = "ConstantFoldingDQ";
+
+    // ConstantFoldingDQ optimization function
+    auto constant_folding_dq_optimization = [&](Graph& graph, const ComputeCapability& optimization_cc, ComputeCapability& cc_to_update) -> Status {
```

Comment:
Can we put this in a separate class/file, maybe in the optimizer library?

Reply:
put it in the new `selection_and_optimization_func.cc/.h`